src/types.ts

```diff
@@ -54,41 +52,26 @@ export type RequestPayloadOf<
 
 type RestrictedUseQueryOptions<
   Response,
-  TError = unknown,
+  TError = DefaultError,
   Data = Response,
 > = Omit<UseQueryOptions<Response, TError, Data>, 'queryKey' | 'queryFn'> & {
   axios?: AxiosRequestConfig;
 };
 
-type RestrictedUseInfiniteQueryOptions<
-  Response,
-  Request,
-  Data = Response,
-> = Omit<
-  UseInfiniteQueryOptions<Response, unknown, Data>,
-  'queryKey' | 'queryFn' | 'getNextPageParam' | 'getPreviousPageParam'
+type RestrictedUseInfiniteQueryOptions<Response, Request> = Omit<
+  UseInfiniteQueryOptions<InfiniteData<Response>, DefaultError>,
+  | 'queryKey'
+  | 'queryFn'
+  | 'initialPageParam'
+  | 'getNextPageParam'
+  | 'getPreviousPageParam'
 > & {
   axios?: AxiosRequestConfig;
-  getNextPageParam?: (lastPage: Response) => Partial<Request> | undefined;
+  initialPageParam: Partial<Request>; // use init payload?
+  getNextPageParam: (lastPage: Response) => Partial<Request> | undefined;
   getPreviousPageParam?: (firstPage: Response) => Partial<Request> | undefined;
 };
 
-type RestrictedUseInfiniteQueryResult<Response, Request> = Omit<
-  UseInfiniteQueryResult<Response>,
-  'fetchNextPage' | 'fetchPreviousPage'
-> & {
-  fetchNextPage: (
-    options?: Omit<FetchNextPageOptions, 'pageParam'> & {
-      pageParam: Partial<Request>;
-    },
-  ) => Promise<InfiniteQueryObserverResult<Response>>;
-  fetchPreviousPage: (
-    options?: Omit<FetchPreviousPageOptions, 'pageParam'> & {
-      pageParam: Partial<Request>;
-    },
-  ) => Promise<InfiniteQueryObserverResult<Response>>;
-};
```

Comment:
Why is this type going away?

Reply:
@swain We no longer need to override any of the types for the result of `useInfiniteAPIQuery`

As part of the v5 migration guide they make it clear that they no longer allow passing pageParam through `fetchNextPage` or `fetchPreviousPage`. [see here](https://tanstack.com/query/v5/docs/framework/react/guides/migrating-to-v5#manual-mode-for-infinite-queries-has-been-removed)

There main reasoning being 

> This feature didn't work at all with refetches and wasn't widely known or used

so we can rely on the default types for these methods